libc/spec/linux.td

```diff
@@ -287,6 +287,23 @@ def Linux : StandardSpec<"Linux"> {
       ]
   >;
 
+
+  HeaderSpec UniStd = HeaderSpec<
+    "unistd.h",
+    [], // Macros
+    [],
+    [], // Enumerations
+    [
+        FunctionSpec<
+          "pipe2",
+          RetValSpec<IntType>,
+          [ArgSpec<IntPtr>] //TODO: make this int[2]
```

Comment:
This needs to be `[ArgSpec<IntPtr>, ArgSpec<IntType>]`. Looks like it's also wrong for `pipe`, but that can be fixed later.
```suggestion
          [ArgSpec<IntPtr>, ArgSpec<IntType>] //TODO: make this int[2]
```

Reply:
Sorry about that. 

 Is there an automated test I should be running which catches discrepancies like this? Or do we just have to be careful here.

Reply:
Also, AFAIK the `pipe` definition in `posix.td` actually doesn't have this problem.

https://github.com/llvm/llvm-project/blob/f54cdc5d6ee5532da117f2489c105148c94dcb39/libc/spec/posix.td#L646-L650

The [posix standard for `pipe`](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/unistd.h.html) defines the signature as...

```
int pipe(int[2]);
```
This matches the `FunctionSpec` we have in `posix.td`.

Reply:
@petrhosek is working on removing the old headergen implementation; we expect it to all go away imminently. So mistakes in the existing TD files are not pressing.